pkg/ovsdb/ovsdb.go

```diff
@@ -329,7 +336,302 @@ func (ovsd *OvsDriver) GetOFPortVlanState(portName string) (string, *uint, []uin
+	portUUID := ovsdb.UUID{GoUUID: portUUIDStr}
+
+	mutateMirrorOp := detachPortFromMirrorOperation(portUUID, mirrorName, MirrorConsumer)
+
+	// Perform OVS transaction
+	operations := []ovsdb.Operation{*mutateMirrorOp}
+
+	_, err := ovsd.ovsdbTransact(operations)
+	return err
+}
+
+// GetMirrorUUID Retrieves the UUID of a mirror from its name
+func (ovsd *OvsBridgeDriver) GetMirrorUUID(mirrorName string) (ovsdb.UUID, error) {
+	condition := ovsdb.NewCondition("name", ovsdb.ConditionEqual, mirrorName)
+	row, err := ovsd.findByCondition("Mirror", condition, nil)
+	if err != nil {
+		return ovsdb.UUID{}, err
+	}
+
+	// We make a select transaction using the interface name
+	// Then get the Mirror UUID from it
+	mirrorUUID := row["_uuid"].(ovsdb.UUID)
+
+	return mirrorUUID, nil
+}
+
+// GetPortUUID Retrieves the UUID of a port from its name
+func (ovsd *OvsBridgeDriver) GetPortUUID(portName string) (ovsdb.UUID, error) {
+	condition := ovsdb.NewCondition("name", ovsdb.ConditionEqual, portName)
+	row, err := ovsd.findByCondition("Port", condition, nil)
+	if err != nil {
+		return ovsdb.UUID{}, err
+	}
+
+	// We make a select transaction using the interface name
+	// Then get the Port UUID from it
+	portUUID := row["_uuid"].(ovsdb.UUID)
+
+	return portUUID, nil
+}
+
+// IsMirrorConsumerAlreadyAttached Checks if the 'output_port' column of a mirror consumer contains a port UUID
+func (ovsd *OvsDriver) IsMirrorConsumerAlreadyAttached(mirrorName string) (bool, error) {
+	condition := ovsdb.NewCondition("name", ovsdb.ConditionEqual, mirrorName)
+	row, err := ovsd.findByCondition("Mirror", condition, nil)
+	if err != nil {
+		return false, err
+	}
+
+	outputPorts, err := convertToArray(row["output_port"])
+	if err != nil {
+		return false, fmt.Errorf("cannot convert output_port to an array error: %v", err)
+	}
+
+	if len(outputPorts) == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+// CheckMirrorProducerWithPorts Checks the configuration of a mirror producer based on ingress and egress values
+func (ovsd *OvsDriver) CheckMirrorProducerWithPorts(mirrorName string, ingress, egress bool, portUUIDStr string) (bool, error) {
+	portUUID := ovsdb.UUID{GoUUID: portUUIDStr}
+
+	var conditions []ovsdb.Condition = []ovsdb.Condition{}
+	conditionName := ovsdb.NewCondition("name", ovsdb.ConditionEqual, mirrorName)
+	conditions = append(conditions, conditionName)
+	if ingress {
+		// select_src_port = Ports on which arriving packets are selected for mirroring
+		conditionIngress := ovsdb.NewCondition("select_src_port", ovsdb.ConditionIncludes, portUUID)
+		conditions = append(conditions, conditionIngress)
+	}
+	if egress {
+		// select_dst_port = Ports on which departing packets are selected for mirroring
+		conditionsEgress := ovsdb.NewCondition("select_dst_port", ovsdb.ConditionIncludes, portUUID)
+		conditions = append(conditions, conditionsEgress)
+	}
+	// We cannot call findByCondition because we need to pass an array of conditions
```

Comment:
Could you define a helper function to do that? So we don't need to duplicate this code here and in the following two functions

Reply:
agree. done